src/demo/App.js

```diff
@@ -1,40 +1,107 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+import TextareaAutosize from 'react-autosize-textarea';
 
 import ReactRRuleGenerator from '../lib';
 import './index.css';
 import githubLogo from './github_logo.png';
 
-const App = () => (
-  <div>
-    <div className="app-navbar">
-      <a href="https://github.com/fafruch/react-rrule-generator">
-        &lt; go back to
-        {' '}
-        <img className="app-navbar-ghlogo" src={githubLogo} alt="Github logo" />
-        /fafruch
-      </a>
-
-      <iframe
-        title="github-star"
-        src="https://ghbtns.com/github-btn.html?user=fafruch&repo=react-rrule-generator&type=star&count=true&size=medium"
-        frameBorder="0"
-        scrolling="0"
-        width="72px"
-        height="20px"
-      />
-    </div>
-    <div className="app-header">
-      <h1>React RRule Generator</h1>
-    </div>
-
-    <div className="app-desc">
-      Recurence rules generator form built with React
-    </div>
-
-    <div className="app">
-      <ReactRRuleGenerator />
-    </div>
-  </div>
-);
+class App extends Component {
+  state = {
+    rrule: '',
+    isCopied: false,
+  };
+
+  handleChange = (newRRule) => {
+    this.setState({ rrule: newRRule, isCopied: false });
+  }
+
+  handleCopy = () => {
+    this.setState({ isCopied: true });
+  }
+
+  render() {
+    const { rrule, isCopied } = this.state;
+
+    return (
+      <div>
+        <div className="app-navbar">
+          <a href="https://github.com/fafruch/react-rrule-generator">
+            &lt; go back to
+            {' '}
+            <img className="app-navbar-ghlogo" src={githubLogo} alt="Github logo" />
+            /fafruch
+          </a>
+
+          <iframe
+            title="github-star"
+            src="https://ghbtns.com/github-btn.html?user=fafruch&repo=react-rrule-generator&type=star&count=true&size=medium"
+            frameBorder="0"
+            scrolling="0"
+            width="72px"
+            height="20px"
+          />
+        </div>
+        <div className="app-header">
+          <h1>React RRule Generator</h1>
+        </div>
+
+        <div className="app-desc">
+          Recurence rules generator form built with React
+        </div>
+
+
+        <div className="app container">
+          <h5><strong>{'<RRuleGenerator />'}</strong></h5>
+
+          <ReactRRuleGenerator
+            onMount={defaultRRule => this.setState({ rrule: defaultRRule })}
+            onChange={newRRule => this.handleChange(newRRule)}
```

Comment:
I done see why onMount would sent value to container. Its the other way round - the container should send value down to the editor i.e. like this:

```
<ReactRRuleGenerator
  value={this.state. rrule}
  onChange={this.handleChange}
/>
```

Reply:
Before I go any further, should default RRule string has an influence on the default view - it means that to configure the view to be adequate to RRule provided and match it 1:1?

Reply:
I think flow should be like this:
- component gests rrule value via `value` prop
- if that value is different the its internal state `rrule` then it: updates it and generates the state based on it

Then, when the rule is changed, you can do the other way round:
- first update internal state (including internal `rrule`)
- then call `onChange` with new `rrule`
This rrule will get to component again via prop, but you will be able to ignore it using  `shouldComponentUpdate` that will compare the `value` and `state.rrule`